### PR TITLE
Charlesmchen/hardware ringer vs playback

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk

### CODE REVIEW

File: Signal/src/ViewControllers/AttachmentApprovalViewController.swift [hunk 1]
```diff
@@ -42,14 +42,26 @@ class AttachmentApprovalViewController: UIViewController {
...
+                                                                target:self,
+                                                                action:#selector(donePressed))
         self.navigationItem.title = NSLocalizedString("ATTACHMENT_APPROVAL_DIALOG_TITLE",
                                                       comment: "Title for the 'attachment approval' dialog.")
 
         createViews()
     }
 
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/AttachmentApprovalViewController.swift]: typo: `super.viewWillAppear`

**Comment by charlesmchen** [comment]
> Does this do the right thing WRT voice notes?

I'm not sure how you mean - we can't record voice notes locally yet.  In many cases where we play or record audio, we update the category of `AVAudioSession` before doing so.  If we're missing that anywhere, that isn't a regression - we're already changing the category in a few places.  Nonetheless the changes here - to make sure we're using the right category during video playback - take care to restore the category to a reasonable default after the playback is done.  I'm using `AVAudioSessionCategoryPlayAndRecord` - but I noticed that CallAudioService sets the category to `AVAudioSessionCategoryAmbient` when a call ends.  

> If you get a call while playing a video it will ring loudly right?

That's `CallAudioService`'s responsibility, and it's `startRinging` method makes sure to use AVAudioSessionCategoryAmbient so that we honor the mute switch.

**Comment by charlesmchen** [comment]
Note that this work doesn't do anything around playback of audio - that's a separate task.

File: Signal/src/ViewControllers/AttachmentApprovalViewController.swift (hunk 1 above)
Comment (charlesmchen) [inline Signal/src/ViewControllers/AttachmentApprovalViewController.swift]: Done.

**Comment by michaelkirk** [comment]
> Note that this work doesn't do anything around playback of audio - that's a separate task.

Ok, that explains my confusion.

**Pull request merged**